File: DSA/binary.cpp
#include<iostream>

using namespace std;

int BinarySearh(int arr[], int n, int target){
    int left  = 0;
    int right = n - 1;

    while(left <= right){
        int mid = left + (right - left) / 2;
        if(arr[mid] == target){
            cout<<"Element found at index: "<< mid << endl;
            return mid;
        }
        else if(arr[mid] < target){
            left = mid + 1;
        }
        else{
            right = mid - 1;
        }
    }
    return -1;



}


int main(){


   int arr[] = {1,2,3,4,5,6,7,8,9};
    int n = sizeof(arr) / sizeof(arr[0]); // ✅ size here
    int target = 5;
   int result = BinarySearh(arr ,n, target);
   printf("Result: %d\n", result);

    return 0;
}

File: DSA/binary.js

nums= [1,2,3,4,5,6,7,8,9,10]
target = 6
const Binary = function(nums , target){
    let n = nums.length;
    let left = 0 
    let right = n-1

    while(left <= right){
        let mid = Math.floor((left +right)/2)
        if(nums[mid] == target){
            return mid

        }
        else if(nums[mid] < target){
            left= mid+1
        }
        else{
            right = mid-1
        }
    }
    return -1

}


console.log(Binary(nums,target))